device/index.js

```diff
@@ -512,7 +512,8 @@ function DeviceClient(options) {
       }
 
       if (!isUndefined(options.host) && isUndefined(options.region)) {
-         var pattern = /[a-zA-Z0-9]+\.iot\.([a-z]+-[a-z]+-[0-9]+)\.amazonaws\..+/;
+         // extract anything in between "iot" and "amazonaws" as region
+         var pattern = /[a-zA-Z0-9]+\.iot\.(.*)\.amazonaws\..+/;
```

Comment:
The region regex can match multiple subdomains or empty string. Maybe we should exclude dots and use the one-or-more quantifier?
```
/[a-zA-Z0-9]+\.iot\.([^.]+)\.amazonaws\..+/
```

Reply:
A valid domain label can't be empty.